source/opt/ssa_rewrite_pass.cpp

```diff
@@ -615,7 +622,11 @@ Pass::Status SSARewriter::RewriteFunctionIntoSSA(Function* fp) {
             << fp->PrettyPrint(0) << "\n";
 #endif
 
-  if (modified) pass_->context()->KillDebugDeclareInsts(fp);
+  if (modified) {
+    for (auto var_id : var_ids_debugdeclare_replaced_) {
```

Comment:
I believe that the target variables (those in `seen_target_vars_`) should be the ones you need to go through.  No need to keep a separate list.  If there is a debug declare for a target variable that is not removed, then I would expect that the be an error.

Could we try to use the target variable list here instead?

Reply:
Thank you for the suggestion.

Initially, I concerned the following case but I agree with you that we can simple check if it is a target variable.
```
 int j; // global variable
 void main()
 {
     float f = 0.0;
     for (j=0; j<4; j++) {
       int& i = j; // I concerned DebugDeclare for i.
       f = f + BC[i];
     }
     fo = f;
 }
```